src/include/codegen/multi_thread_context.h

```diff
@@ -24,18 +26,46 @@ namespace codegen {
 class MultiThreadContext {
  public:
 
-  MultiThreadContext(const CodeGen &codegen, uint64_t num_threads): codegen_(codegen) {
-      num_threads_ = codegen_.Const64(num_threads);
+  MultiThreadContext(llvm::Value *thread_id, llvm::Value *num_threads, std::atomic<bool> *finished)
```

Comment:
will there be any problem, if the thread_id here is larger than the actual number of threads executing this query?